### PR TITLE
Optimisations phase 1

### PR DESCRIPTION
**Pull request by lupin012**
This branch contains few optimizations removed few  pair copy

### CODE REVIEW

File: tests/perf/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ ulimit -c 3000000
 # to increase max file descriptors
 ulimit -n 64000
 # to increase port list (sudo su) 
-echo 10000 61000 > /proc/sys/net/ipv4/ip_local_port_range
+echo 5000 65000 > /proc/sys/net/ipv4/ip_local_port_range
```
Comment (canepat) [inline tests/perf/README.md:10]: It seems that the command above expands the range temporarily and the following is necessary to make the change persistent:
```
sudo sysctl -n net.ipv4.ip_local_port_range="5000 65000"
```
Please check this again and if so add to README

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkrpc/CMakeLists.txt [hunk 2]
```diff
@@ -158,12 +156,13 @@ target_link_libraries(silkrpc
...
+    /usr/local/lib/libmimalloc.so)
 
 add_executable(silkrpcdaemon main.cpp)
 target_include_directories(silkrpcdaemon PUBLIC ${CMAKE_SOURCE_DIR})
 target_link_libraries(silkrpcdaemon
     silkrpc
     absl::flags_parse
     silkinterfaces
-    tcmalloc)
+    /usr/local/lib/libmimalloc.so)
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkrpc/CMakeLists.txt:168]: ok

File: silkrpc/CMakeLists.txt [hunk 3]
```diff
@@ -158,12 +156,13 @@ target_link_libraries(silkrpc
     protobuf::libprotobuf
     silkinterfaces
     silkworm_core
-    silkworm_node)
+    silkworm_node
+    /usr/local/lib/libmimalloc.so)
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkrpc/CMakeLists.txt:160]: ok

File: tests/perf/README.md (hunk 1 above)
Comment (lupin012) [reply tests/perf/README.md]: ok to be permanent should be written also in /etc/sysctl.conf

**Review by canepat — APPROVED** [review]

**Pull request merged**